courses/utils.py

```diff
@@ -0,0 +1,251 @@
+        #       of this method so we don't need to worry about user=None here.
+        if (
+            CourseRunGrade.objects.filter(user=user, course_run=course_run).exists()
+            and not force
+        ):
+            logger.warning(
+                "Grade exists for user: %s and courseware_id: %s. Not overwriting",
+                user.email,
+                course_run.courseware_id,
+            )
+            return
+        # Forced overwrite or record does not exist already
+        CourseRunGrade.objects.update_or_create(
+            user=user,
+            course_run=course_run,
+            defaults={
+                "grade": override_grade,
+                "passed": bool(override_grade),
+                "letter_grade": None,
+            },
+        )
+
+
+def sync_courserun_certificates(readable_id, email=None):
+    """
+    Create course run certificates
+    """
+    user = None
+    course_run = CourseRun.objects.filter(courseware_id=readable_id).first()
+
+    if not course_run:
+        exception = CourseRun.DoesNotExist(
+            "Could not find course run with courseware_id={}".format(readable_id)
+        )
+        logger.error(str(exception))
+        raise exception
+
+    if email:
+        user = fetch_user(email)
+
+    # Generate certificate for a specific user
+    if user:
+        run_grade = None
+        try:
+            run_grade = CourseRunGrade.objects.get(course_run=course_run, user=user)
+        except CourseRunGrade.DoesNotExist:
+            exception = CourseRunGrade.DoesNotExist(
+                "No grade exists for user with email={} for courseware_id={}. Cannot generate certificate.".format(
+                    user.email, course_run.courseware_id
+                )
+            )
+            logger.error(str(exception))
+            raise exception
+
+        _generate_certificate(run_grade)
+    # Generate certificates for all users enrolled in the course run
+    else:
+        run_grades = CourseRunGrade.objects.filter(course_run=course_run)
+        for run_grade in run_grades:
+            _generate_certificate(run_grade)
+
+
+def _generate_certificate(run_grade):
+    """
+    Generates a certificate based on the CourseRunGrade
+    """
+    # Has a passing grade
+    if run_grade.grade:
+        if CourseRunCertificate.objects.filter(
+            user=run_grade.user, course_run=run_grade.course_run
+        ).exists():
+            logger.warning(
+                "Certificate already exists for user with email=%s for courseware_id=%s. Not regenerating.",
+                run_grade.user.email,
+                run_grade.course_run.courseware_id,
+            )
+        else:
+            logger.info(
+                "Generating certificate for user with email=%s and courseware_id=%s.",
+                run_grade.user.email,
+                run_grade.course_run.courseware_id,
+            )
+            CourseRunCertificate.objects.create(
```

Comment:
why not we just simply call the `get_or_create` instead of these if conditions ?